transcom/mymove
# B 19888 service item estimate prices int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19888)

## Summary

When the prime provides an estimated weight, we want to calculate the estimated price and store it on the service item in the `pricing_estimate` column.

To test:
1. Run `make tasks_save_ghc_fuel_price_data`
2. Create a shipment in the prime sim, and make a note of your requested pickup date and your estimated price.
3. Approve the shipment as the TOO.
4. Check the mto_service_items table and verify that prices exist for the initial 6 service items that get created upon TOO approval.
5. Go back to the prime sim
6. Update the shipment to have the same `actual` pickup and `actual` weight that you put down for the requested pickup and estimated weight.
7. After updating, create a payment request as the prime sim.
8. Make sure you use the shipment you created and updated - then add all the service items and submit the payment request.
9. As the TIO, review the payment request and compare the prices on that page to the values in the database.

A note on the tests - there is quite a bit of test setup data required to have all the inputs for every pricing function. The pricing functions I utilized already have pretty extensive tests written for them, so I don't see a need to retest them in the shipment approver tests.

![image](https://github.com/transcom/mymove/assets/147537467/20c706dd-1bbb-4488-a456-cff1f574ec72)

![image](https://github.com/transcom/mymove/assets/147537467/084e5b8f-7f8b-4405-bbc2-c9499cba2d9f)
